docs/language.md

```diff
@@ -154,28 +225,52 @@ the `prior()` function:
     prior(this)         - Value of `this` before the write is completed.
     prior(this.prop)    - Value of a property before the write is completed.
 
-You can also use `prior()` to wrap any expressions (including function calls) that
+`prior()` can be used to wrap any expressions (including function calls) that
 use `this`.
 
+The parent key of the current location can be read using the key() function.
+
+    key()               - The (text) value of the inner-most parent property of the current location.
+
+This can be used to create a validation expression that relates the key used to store a value
+and one of its properties:
+
+```javascript
+path /products is Product[];
+
+type Product {
+  validate() = this.id == key();
+
+  id: String,
+  name: String
+}
+
+```
+
+
 # Functions and Methods
 
 Functions must be simple return expressions with zero or more parameters.  All of the following
 examples are identical and can be used interchangably.
 
-    function myFunction(arg1, arg2) {
-      return arg1 == arg2.value;
-    }
+```javascript
+function myFunction(arg1, arg2) {
```

Comment:
can we use a better name and a real example, maybe auth as thats the best use of functions